src/docs/arch-block.dox

```diff
@@ -1,9 +1,207 @@
+metadata file. However the block manager itself isn't created. Only when a btree
+is opened does WiredTiger allocate a block manager. Each btree has its own
+block manager.
+
+@subsection read Reading files and pages
+
+When an existing btree is opened for the first time, the location of the root
+block is contained in the metadata file \c WiredTiger.wt. The block manager will
+read the block at the location specified and return the page image as a buffer
+to the layer above. This will then be instantiated as a page in memory.
+
+From there subsequent page addresses can be read from the root page and the
+process repeated as required. If a cursor traverses to a page which hasn't been
+read into memory the same process will take place.
+
+@subsection Writing
+
+Two cases exist for writing out data using the block manager: checkpoint and
+eviction. When a page image is written out the block manager the \c bm->write
+API is called. See \c bt_io.c for more detail.
+
+@subsubsection Checkpoint
+
+For details on checkpoint at the WiredTiger level see @ref arch-checkpoint.
+
+At the block manager level, a checkpoint corresponds with a set of blocks that
+are stored in the file associated with the given URI. Typically each file will
+contain a minimum of two checkpoints. Upon opening an existing file the most
+recent checkpoint is read.
+
+During a checkpoint new blocks are only written out for dirty pages. A block can
+be included in multiple checkpoints. Assuming a page \c X is dirty and gets
+checkpointed in checkpoint \c A, it will be created as a new block on disk. Now
+the same page \c X isn't modified and another checkpoint is taken. The page is
+clean and as such will not require a new block to be written for it. The address
+of the original block is still valid.
+
+Checkpoints are created in depth first order, leaf blocks are created, then
+the parent blocks. This is a requirement as the parent blocks contain the
+addresses of the leaf blocks.
+
+The block manager doesn't guarantee that calling \c bm->write will result in
+the data being flushed to disk. In the checkpoint scenario WiredTiger will also
+call \c bm->sync which will call the file system dependent flush function.
+
+As a checkpoint progresses it updates its three extent lists, these extent lists
+are written out to disk as part of the checkpoint in blocks. When a checkpoint
+is written the previous checkpoints will be cleaned up if possible, suppose
+the block manager has finished writing out a checkpoint. Which we will call
+checkpoint \c C, checkpoints \c A and \c B exist on disk. By default the block
+manager only retains two checkpoints so it can cleanup checkpoint \c A.
+Checkpoint \c A's extent lists get merged into checkpoint \c B's extent lists.
+If a block, \c N, is created in checkpoint \c A and removed in checkpoint \c B
+the space occupied by the block is considered free and can be utilized.
+
+@subsubsection Eviction
```

Comment:
link to the Eviction page.  Sorry, I meant to flag this in my last review.

Reply:
There isn't an eviction page as far as I can tell?

Reply:
It hasn't been filled in yet, but the stub page is there.  I figured if you put in a link now, we won't have to go back and update this once the eviction pieces gets written.  I suppose an alternative would be to leave a comment on WT-6909 asking whoever writes that to add a link here when it's done.

Reply:
Added in: 9d223fd